src/jdk.jdwp.agent/share/native/libjdwp/util.c

```diff
@@ -504,13 +506,42 @@ sharedGetFieldValues(PacketInputStream *in, PacketOutputStream *out,
         return;
     }
 
-    WITH_LOCAL_REFS(env, length + 1) { /* +1 for class with instance fields */
+    fieldIDs = jvmtiAllocate(length * (jint)sizeof(*fieldIDs));
+    if (fieldIDs == NULL && length > 0) {
+        outStream_setError(out, JDWP_ERROR(OUT_OF_MEMORY));
+        return;
+    }
 
-        int i;
+    if (!isStatic) {
+        clazz = JNI_FUNC_PTR(env,GetObjectClass)(env, object);
```

Comment:
You need to free the jobject before exiting this function. Note you only do this if clazz is allocated here. The assignment to clazz above is from a jobject that the debug agent is managing (as part of mapping JDWP ObjectIDs to jobjects), so it should not be freed for the static case.